docs/STYLEGUIDE.md

```diff
@@ -0,0 +1,208 @@
+
+**When generating or reviewing code, LLM should:**
+
+1. **Detect export keywords** - Scan for `export function`, `export const`, `export type`, `export interface`
+2. **Check for JSDoc** - Verify each export has proper `/** ... */` documentation
+3. **Suggest JSDoc format** - Auto-complete JSDoc blocks for any missing documentation
+4. **Flag regular comments** - Convert `//` comments above exports to JSDoc format
+5. **Apply to all files** - Enforce in `apps/`, `packages/`, and all TypeScript files
+
+### Detection Patterns for Regular Comments Above Exports
+
+**🚨 CRITICAL: Always detect and flag these patterns for JSDoc conversion:**
+
+- `// comment\nexport function` → Convert to JSDoc
+- `/* comment */\nexport function` → Convert to JSDoc
+- `// comment\nexport const` → Convert to JSDoc
+- `// comment\nexport interface` → Convert to JSDoc
+- `// comment\nexport type` → Convert to JSDoc
+- `// comment\nexport class` → Convert to JSDoc
+
+**Example:**
+
+❌ **Avoid - Regular comment above export:**
+
+```typescript
+// calculates position notional value
+export function calculateNotional(size: BigDecimal, price: BigDecimal) {
+  // Should be converted to JSDoc format
+}
+```
+
+✅ **Good - Proper JSDoc:**
+
+```typescript
+/**
+ * Calculates the notional value of a position
+ * @param size - Position size in base units (BigDecimal)
+ * @param price - Current price per unit (BigDecimal, precision 18)
+ * @returns Notional value in quote currency (USDC), rounded to 6 decimal places
+ * @throws {InvalidPositionError} When size is zero or negative
+ */
+export function calculateNotional(
+  size: BigDecimal,
+  price: BigDecimal,
+): BigDecimal {
+  // Implementation
+}
+```
+
+## TypeScript Conventions
+
+- Use `interface` for object shapes that might be extended or implemented
+- Use `type` for unions, primitives, computed types, and utility types
+- **Never use `any` type** - Prefer `unknown` for truly unknown types, or create proper type definitions
+- Use descriptive generic constraints: `<T extends Record<string, unknown>>`
+
+## Class and Method Patterns
+
+✅ **Good client class pattern:**
+
+```typescript
+/**
+ * Main client for interacting with Vertex Protocol
+ */
+export class VertexClient {
+  private readonly config: VertexClientConfig;
+  private readonly signer?: Signer;
+
+  /**
+   * Creates a new Vertex client instance
+   * @param config - Client configuration options
+   * @param signer - Optional signer for authenticated operations
+   */
+  constructor(config: VertexClientConfig, signer?: Signer) {
+    this.config = config;
+    this.signer = signer;
+  }
+
+  /**
+   * Retrieves current market data for a product
+   * @param productId - The product ID to query
+   * @returns Promise resolving to market data
+   * @throws {ProductNotFoundError} When product ID is invalid
+   */
+  async getMarketData(productId: number): Promise<MarketData> {
+    // Implementation
+  }
+}
+```
+
+## Error Handling Patterns
```

Comment:
Hmm, we don't have a `VertexError`

Reply:
Adjusted all examples to use what's within SDK.